misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt

```diff
@@ -81,22 +81,30 @@ internal class RealTransacter private constructor(
     while (true) {
       try {
         attempt++
-        return transactionInternal(block)
+        val result = transactionInternal(block)
+
+        if (attempt > 0) {
```

Comment:
Seems like this is always true. Maybe `attempt > 1`?

Reply:
Good catch!